Add XDG_CONFIG_HOME compatability

tomswartz07 opened the pull request:
## Background
The currently implemented config dir search paths do not follow Linux XDG standards.

## Implementation
This PR enables the program to search in the standard `$(HOME)/.config/` path, according to [XDG Standards](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).

This PR simply adds an additional path on top of the currently defined config dirs, so there is little chance for regression if the other paths were used.

------ 

Thanks for a great program! Hope to continue contributing where possible in the future!

zuckschwerdt commented:
Thanks. Looks good. You can remove/replace the `~/.rtl_433/rtl_433.conf` path. Better get it right now. Also expand `XDG_CONFIG_HOME` or even add a search for `XDG_CONFIG_DIRS` if you like.

tomswartz07 commented:
@zuckschwerdt Done and done. Dropped the old `~/.rtl_433...` path and replaced it with a check for the `XDG_CONFIG_HOME` var, failing back to the standard location if it's not set.

Thanks!

tomswartz07 commented:
Ah, that's a much more clean way of approaching it.
I'll implement is as suggested.
